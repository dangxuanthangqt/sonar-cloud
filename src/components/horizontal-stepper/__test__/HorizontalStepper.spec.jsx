import React from 'react';
import {
  getByRole,
  render,
  screen,
  waitFor,
} from '../../../common/utils/testUtils';
import HorizontalLinearStepper from '../index';

describe('Horizontal stepper', () => {
  it('should match snapshot', async () => {
    const { container } = render(<HorizontalLinearStepper />);
    expect(container).toMatchSnapshot();
  });
  it('should match element in document', async () => {
    render(<HorizontalLinearStepper />);

    expect(screen.getByRole('button', { name: '' })).toBeInTheDocument();

    expect(screen.getByText('Previous')).toBeInTheDocument();

    expect(screen.getByText('Datasources')).toBeInTheDocument();

    expect(screen.getByText('Keywords')).toBeInTheDocument();

    expect(screen.getByText('Vehicles')).toBeInTheDocument();

    expect(screen.getByText('Sales Codes')).toBeInTheDocument();

    expect(screen.getByText('LOPs Parts')).toBeInTheDocument();

    expect(screen.getByText('Summary')).toBeInTheDocument();
  });
});
