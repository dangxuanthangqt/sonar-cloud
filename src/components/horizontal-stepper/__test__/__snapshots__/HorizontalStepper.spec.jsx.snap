// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Horizontal stepper should match snapshot 1`] = `
<div>
  <div
    class="makeStyles-container-1 MuiBox-root css-0"
  >
    <div
      class="flex justify-center flex-col items-center"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium makeStyles-iconPrevious-8 css-v3imdt-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17slstj-MuiSvgIcon-root"
          data-testid="NavigateBeforeRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M14.91 6.71a.9959.9959 0 0 0-1.41 0L8.91 11.3c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L11.03 12l3.88-3.88c.38-.39.38-1.03 0-1.41z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <p
        class="makeStyles-titleBtnPrevious-9"
        style="display: none;"
      >
        Previous
      </p>
    </div>
    <div
      class="MuiStepper-root MuiStepper-horizontal makeStyles-stepper-10 css-m5vj9m-MuiStepper-root"
    >
      <div
        class="MuiStep-root MuiStep-horizontal makeStyles-step-11 css-1bw0nnu-MuiStep-root"
      >
        <span
          class="makeStyles-stepLabelRoot-2 MuiStepLabel-root MuiStepLabel-horizontal css-ascpo7-MuiStepLabel-root"
        >
          <span
            class="makeStyles-iconStyle-3 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root Mui-active css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                dominant-baseline="central"
                text-anchor="middle"
                x="12"
                y="12"
              >
                1
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
          >
            <span
              class="makeStyles-labelStyle1-4 MuiStepLabel-label makeStyles-activeStep-5 Mui-active css-gv1taf-MuiStepLabel-label"
            >
              Datasources
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal makeStyles-step-11 css-1bw0nnu-MuiStep-root"
      >
        <span
          class="makeStyles-stepLabelRoot-2 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
        >
          <span
            class="makeStyles-iconStyle-3 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                dominant-baseline="central"
                text-anchor="middle"
                x="12"
                y="12"
              >
                2
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
          >
            <span
              class="makeStyles-labelStyle1-4 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
            >
              Keywords
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal makeStyles-step-11 css-1bw0nnu-MuiStep-root"
      >
        <span
          class="makeStyles-stepLabelRoot-2 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
        >
          <span
            class="makeStyles-iconStyle-3 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                dominant-baseline="central"
                text-anchor="middle"
                x="12"
                y="12"
              >
                3
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
          >
            <span
              class="makeStyles-labelStyle1-4 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
            >
              Vehicles
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal makeStyles-step-11 css-1bw0nnu-MuiStep-root"
      >
        <span
          class="makeStyles-stepLabelRoot-2 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
        >
          <span
            class="makeStyles-iconStyle-3 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                dominant-baseline="central"
                text-anchor="middle"
                x="12"
                y="12"
              >
                4
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
          >
            <span
              class="makeStyles-labelStyle1-4 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
            >
              Sales Codes
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal makeStyles-step-11 css-1bw0nnu-MuiStep-root"
      >
        <span
          class="makeStyles-stepLabelRoot-2 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
        >
          <span
            class="makeStyles-iconStyle-3 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                dominant-baseline="central"
                text-anchor="middle"
                x="12"
                y="12"
              >
                5
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
          >
            <span
              class="makeStyles-labelStyle1-4 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
            >
              Dates
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal makeStyles-step-11 css-1bw0nnu-MuiStep-root"
      >
        <span
          class="makeStyles-stepLabelRoot-2 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
        >
          <span
            class="makeStyles-iconStyle-3 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                dominant-baseline="central"
                text-anchor="middle"
                x="12"
                y="12"
              >
                6
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
          >
            <span
              class="makeStyles-labelStyle1-4 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
            >
              LOPs Parts
            </span>
          </span>
        </span>
      </div>
      <div
        class="MuiStepConnector-root MuiStepConnector-horizontal Mui-disabled css-j5w0w9-MuiStepConnector-root"
      >
        <span
          class="MuiStepConnector-line MuiStepConnector-lineHorizontal css-z7uhs0-MuiStepConnector-line"
        />
      </div>
      <div
        class="MuiStep-root MuiStep-horizontal makeStyles-step-11 css-1bw0nnu-MuiStep-root"
      >
        <span
          class="makeStyles-stepLabelRoot-2 MuiStepLabel-root MuiStepLabel-horizontal Mui-disabled css-ascpo7-MuiStepLabel-root"
        >
          <span
            class="makeStyles-iconStyle-3 MuiStepLabel-iconContainer css-vnkopk-MuiStepLabel-iconContainer"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiStepIcon-root css-1ktinih-MuiSvgIcon-root-MuiStepIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <circle
                cx="12"
                cy="12"
                r="12"
              />
              <text
                class="MuiStepIcon-text css-ighunu-MuiStepIcon-text"
                dominant-baseline="central"
                text-anchor="middle"
                x="12"
                y="12"
              >
                7
              </text>
            </svg>
          </span>
          <span
            class="MuiStepLabel-labelContainer css-16ubnlw-MuiStepLabel-labelContainer"
          >
            <span
              class="makeStyles-labelStyle1-4 MuiStepLabel-label Mui-disabled css-gv1taf-MuiStepLabel-label"
            >
              Summary
            </span>
          </span>
        </span>
      </div>
    </div>
    <div>
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium makeStyles-iconBtnNext-14 css-1vmeuix-MuiButtonBase-root-MuiIconButton-root"
        tabindex="0"
        type="button"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-17slstj-MuiSvgIcon-root"
          data-testid="NavigateNextRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9.31 6.71c-.39.39-.39 1.02 0 1.41L13.19 12l-3.88 3.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41L10.72 6.7c-.38-.38-1.02-.38-1.41.01z"
          />
        </svg>
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <p
        class="makeStyles-titleBtnNext-13"
        style="color: rgb(18, 41, 62);"
      >
        Next
      </p>
    </div>
  </div>
</div>
`;
